File: hw/pilatasig/u2/HW_18_Exeptions/MoleProjectManager - 1.1.3/src/main/java/utils/MongoDBConnection.java
package utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
/**
 *
 * @author Usuario
 */
public class MongoDBConnection {
    private static final String CONNECTION_STRING = "mongodb+srv://dapilatasig:dapilatasig@cluster0.1qaby.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
    private static final String DATABASE_NAME = "OOP_MoleProject";
    private static MongoDatabase database;

    public static MongoDatabase getDatabase() {
        if (database == null) {
            MongoClient mongoClient = MongoClients.create(CONNECTION_STRING);
            database = mongoClient.getDatabase(DATABASE_NAME);
        }
        return database;
    }
}

File: hw/pazmino/u3/HW30URIs/hw40URIPython/utils/mongodb_connection.py
import os
from pymongo import MongoClient

class MongoDBConnection:
    CONNECTION_STRING = "mongodb+srv://dapilatasig:dapilatasig@cluster0.1qaby.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
    DATABASE_NAME = "OOP_MoleProject"
    COLLECTION_NAME = "Projects"
    _database = None

    @classmethod
    def get_database(cls):
        if cls._database is None:
            mongo_client = MongoClient(cls.CONNECTION_STRING)
            cls._database = mongo_client[cls.DATABASE_NAME]
        return cls._database
